controllers/restic.go

```diff
@@ -82,40 +82,39 @@ func (r *VolumeSnapshotBackupReconciler) CreateVSBResticSecret(log logr.Logger)
 	var scheduleCronExpr = ""
 	rpolicy := RetainPolicy{}
 	for key, val := range resticSecret.Data {
+		stringVal := string(val)
 		if key == ResticRepository {
 			// if trailing '/' in user-created Restic repo, remove it
-			stringVal := string(val)
-			stringVal = strings.TrimRight(stringVal, "/")
-
-			ResticRepoValue = stringVal
+			ResticRepoValue = strings.TrimRight(stringVal, "/")
 		}
 		if key == ResticPruneInterval {
-			pruneInterval = string(val)
+			pruneInterval = stringVal
 		}
 		if key == SnapshotScheduleCron {
-			scheduleCronExpr = string(val)
+			scheduleCronExpr = stringVal
 		}
 		if key == SnapshotRetainPolicyMonthly {
-			rpolicy.monthly = string(val)
+			rpolicy.monthly = stringVal
 		}
 		if key == SnapshotRetainPolicyDaily {
-			rpolicy.daily = string(val)
+			rpolicy.daily = stringVal
 		}
 		if key == SnapshotRetainPolicyHourly {
-			rpolicy.hourly = string(val)
+			rpolicy.hourly = stringVal
 		}
 		if key == SnapshotRetainPolicyWeekly {
-			rpolicy.weekly = string(val)
+			rpolicy.weekly = stringVal
 		}
 		if key == SnapshotRetainPolicyYearly {
-			rpolicy.yearly = string(val)
+			rpolicy.yearly = stringVal
 		}
 		if key == SnapshotRetainPolicyWithin {
-			rpolicy.within = string(val)
+			rpolicy.within = stringVal
 		}
 	}
 
-	resticrepo := fmt.Sprintf("%s/%s/%s", ResticRepoValue, pvc.Namespace, pvc.Name)
+	// TODO should check if label is present first? or it is always present?
+	resticrepo := fmt.Sprintf("%s/%s/%s/%s", ResticRepoValue, pvc.Namespace, vsb.Labels["velero.io/backup-name"], pvc.Name)
```

Comment:
would be interesting to change  VSCName-pvc to PVCName-pvc? (the variable names in the code are misleading :exploding_head: )
Ex.: from the bookbag example
```
$ oc get pvc -n minimal-3csivol
NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                 AGE
volume1   Bound    pvc-aaabf682-25c2-4b5c-8795-0ef03e572a6e   10Gi       RWO            ibmc-vpc-block-10iops-tier   48s
volume2   Bound    pvc-a54474ac-e779-40b3-83dd-0fa50e44234c   10Gi       RWO            ibmc-vpc-block-10iops-tier   48s
volume3   Bound    pvc-a08692d8-e421-42af-bdc7-448ba7e172a7   10Gi       RWO            ibmc-vpc-block-10iops-tier   48s
$ oc get vsc
NAME                                               READYTOUSE   RESTORESIZE   DELETIONPOLICY   DRIVER                 VOLUMESNAPSHOTCLASS      VOLUMESNAPSHOT         VOLUMESNAPSHOTNAMESPACE   AGE
snapcontent-8281b69d-f406-486e-8a68-40996de22e4e   true         1073741824    Retain           vpc.block.csi.ibm.io   oadp-example-snapclass   velero-volume1-drcdq   minimal-3csivol           45m
snapcontent-a138c84d-fcf3-4dbd-8312-a8670c13a2a5   true         1073741824    Retain           vpc.block.csi.ibm.io   oadp-example-snapclass   velero-volume3-8hvpm   minimal-3csivol           44m
snapcontent-b5511d2d-6a13-4a36-8d44-dec9b2c908cc   true         1073741824    Retain           vpc.block.csi.ibm.io   oadp-example-snapclass   velero-volume2-kp2r7   minimal-3csivol           44m
velero-velero-volume1-drcdq-bbs9g                  true         1073741824    Retain           vpc.block.csi.ibm.io   oadp-example-snapclass   velero-volume1-drcdq   minimal-3csivol           4m47s
velero-velero-volume2-kp2r7-55v84                  true         1073741824    Retain           vpc.block.csi.ibm.io   oadp-example-snapclass   velero-volume2-kp2r7   minimal-3csivol           4m44s
velero-velero-volume3-8hvpm-dqb4m                  true         1073741824    Retain           vpc.block.csi.ibm.io   oadp-example-snapclass   velero-volume3-8hvpm   minimal-3csivol           4m28s
```
with the change, it would be saved as 
`bucketName/OADPNameSpace/BackupName/snapcontent-8281b69d-f406-486e-8a68-40996de22e4e-pvc/files` 
`bucketName/OADPNameSpace/BackupName/snapcontent-a138c84d-fcf3-4dbd-8312-a8670c13a2a5-pvc/files`
`bucketName/OADPNameSpace/BackupName/snapcontent-b5511d2d-6a13-4a36-8d44-dec9b2c908cc-pvc/files`

Would it be better to be 
`bucketName/OADPNameSpace/BackupName/volume1-pvc/files`
`bucketName/OADPNameSpace/BackupName/volume2-pvc/files`
`bucketName/OADPNameSpace/BackupName/volume3-pvc/files`?

Reply:
would be easier for admins perhaps to review.. but more chance for conflicts since backups run on a schedule will frequently write to the same path. which could be a good thing for deduplication etc.